src/types/function.rs

```diff
@@ -217,6 +242,34 @@ where
     )
 }
 
+unsafe extern "C" fn run_closure_bound<'py, F, R>(
+    capsule_ptr: *mut ffi::PyObject,
+    args: *mut ffi::PyObject,
+    kwargs: *mut ffi::PyObject,
+) -> *mut ffi::PyObject
+where
+    F: Fn(Bound<'py, PyTuple>, Option<Bound<'py, PyDict>>) -> R + Send + 'static,
+    R: crate::callback::IntoPyCallbackOutput<*mut ffi::PyObject>,
+{
+    use crate::types::any::PyAnyMethods;
+
+    crate::impl_::trampoline::cfunction_with_keywords(
+        capsule_ptr,
+        args,
+        kwargs,
+        |py, capsule_ptr, args, kwargs| {
+            let boxed_fn: &ClosureDestructor<F> =
+                &*(ffi::PyCapsule_GetPointer(capsule_ptr, closure_capsule_name().as_ptr())
+                    as *mut ClosureDestructor<F>);
+            let args = Bound::from_borrowed_ptr(py, args).downcast_into_unchecked::<PyTuple>();
+            let kwargs = Bound::from_borrowed_ptr_or_opt(py, kwargs)
+                .map(|b| b.downcast_into_unchecked::<PyDict>());
+            let result = (boxed_fn.closure)(args, kwargs);
```

Comment:
This line is currently giving this error:

```rust
error[E0521]: borrowed data escapes outside of closure
   --> src/types/function.rs:267:26
    |
245 | unsafe extern "C" fn run_closure_bound<'py, F, R>(
    |                                        --- lifetime `'py` defined here
...
260 |         |py, capsule_ptr, args, kwargs| {
    |          --
    |          |
    |          `py` is a reference that is only valid in the closure body
    |          has type `marker::Python<'1>`
...
267 |             let result = (boxed_fn.closure)(args, kwargs);
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                          |
    |                          `py` escapes the closure body here
    |                          argument requires that `'1` must outlive `'py`
```

I'm not clear how to resolve this. Am I even on the right track with my other changes?